conans/test/integration/toolchains/meson/test_mesontoolchain.py

```diff
@@ -205,3 +206,80 @@ def test_clang_cl_vscrt(build_type, runtime, vscrt):
     t.run("install . -pr:h=profile -pr:b=profile")
     content = t.load(MesonToolchain.native_filename)
     assert f"b_vscrt = '{vscrt}'" in content
+
+
+def test_env_vars_from_build_require():
+    conanfile = textwrap.dedent("""
+    from conan import ConanFile
+    import os
+
+    class HelloConan(ConanFile):
+        name = 'hello_compiler'
+        version = '1.0'
+        def package_info(self):
+            self.buildenv_info.define("CC", "CC_VALUE")
+            self.buildenv_info.define("CC_LD", "CC_LD_VALUE")
+            self.buildenv_info.define("CXX", "CXX_VALUE")
+            self.buildenv_info.define("CXX_LD", "CXX_LD_VALUE")
+            self.buildenv_info.define("AR", "AR_VALUE")
+            self.buildenv_info.define("STRIP", "STRIP_VALUE")
+            self.buildenv_info.define("AS", "AS_VALUE")
+            self.buildenv_info.define("WINDRES", "WINDRES_VALUE")
+            self.buildenv_info.define("PKG_CONFIG", "PKG_CONFIG_VALUE")
+            self.buildenv_info.define("LD", "LD_VALUE")
+    """)
+    client = TestClient()
+    client.save({"conanfile.py": conanfile})
+    client.run("create .")
+
+    conanfile = textwrap.dedent("""
+    from conan import ConanFile
+    class HelloConan(ConanFile):
+        name = 'consumer'
+        version = '1.0'
+        generators = "MesonToolchain"
+        settings = "os", "arch", "compiler", "build_type"
+
+        def requirements(self):
+            self.tool_requires("hello_compiler/1.0", )
+    """)
+    # Now, let's check how all the build env variables are applied at consumer side
+    client.save({"conanfile.py": conanfile})
+    client.run("install . -pr:h=default -pr:b=default")
+    content = client.load("conan_meson_native.ini")
+    assert "c = 'CC_VALUE'" in content
+    assert "cpp = 'CXX_VALUE'" in content
+    assert "ld = 'LD_VALUE'" in content
+    assert "c_ld = 'CC_LD_VALUE'" in content
+    assert "cpp_ld = 'CXX_LD_VALUE'" in content
+    assert "ar = 'AR_VALUE'" in content
+    assert "strip = 'STRIP_VALUE'" in content
+    assert "as = 'AS_VALUE'" in content
+    assert "windres = 'WINDRES_VALUE'" in content
+    assert "pkgconfig = 'PKG_CONFIG_VALUE'" in content
```

Comment:
This test is coming from the `test_meson_build_require`. Previously, it was functional, but it shouldn't.